api/runnerpool/naive_placer.go

```diff
@@ -52,16 +58,43 @@ func (sp *naivePlacer) PlaceCall(rp RunnerPool, ctx context.Context, call Runner
 					log.WithError(err).Error("Failed during call placement")
 				}
 				if placed {
+					if err != nil {
+						stats.Record(ctx, placedErrorCountMeasure.M(0))
+					} else {
+						stats.Record(ctx, placedOKCountMeasure.M(0))
+					}
+					// IMPORTANT: here we use (attempt_time - start_time). We want to exclude TryExec
+					// latency *if* TryExec() goes through with success. Placer latency metric only shows
+					// how much time are spending in Placer loop/retries. The metric includes rtt/latency of
+					// *all* unsuccessful NACK (retriable) responses from runners as well. For example, if
+					// Placer loop here retries 4 runners (which takes 5 msecs each) and then 5th runner
+					// succeeds (but takes 35 seconds to finish execution), we report 20 msecs as our LB
+					// latency.
+					stats.Record(ctx, placerLatencyMeasure.M(int64(attemptTime.Sub(startTime)/time.Millisecond)))
 					return err
 				}
+
+				// Too Busy is super common case, we track it separately
+				if err == models.ErrCallTimeoutServerBusy {
+					stats.Record(ctx, retryTooBusyCountMeasure.M(0))
+				} else {
+					stats.Record(ctx, retryErrorCountMeasure.M(0))
+				}
 			}
 		}
 
+		stats.Record(ctx, fullScanCountMeasure.M(0))
+
 		// backoff
 		select {
 		case <-ctx.Done():
-			return models.ErrCallTimeoutServerBusy
+			break OutTries
 		case <-time.After(sp.rrInterval):
 		}
 	}
+
+	// Cancel Exit Path / Client cancelled/timedout
+	stats.Record(ctx, cancelCountMeasure.M(0))
+	stats.Record(ctx, placerLatencyMeasure.M(int64(time.Now().Sub(startTime)/time.Millisecond)))
+	return models.ErrCallTimeoutServerBusy
```

Comment:
And, same question here about the both the recording of stats and return of `ErrCallTimeoutServerBusy` in the case where no servers are found.